flocker/control/httpapi.py

```diff
@@ -351,7 +354,7 @@ def saved(ignored):
                       '/v1/endpoints.json#/definitions/configuration_dataset'},
         schema_store=SCHEMAS
     )
-    def update_dataset(self, dataset_id, primary=None):
+    def update_dataset(self, dataset_id, primary=None, maximum_size=None):
```

Comment:
Will this work support removing a maximum size from a ZFS-backed dataset?  I think that's represented as a `maximum_size` of `None` in the current configuration model.

I think it would be okay to defer this work until later.  Resizing ZFS will work without it, it just won't be possible to express the change from a size limit to no size limit - which is probably not an important limitation in the near term.  But file an issue for addressing this later.